pkg/autoscaling/calculation.go

```diff
@@ -103,14 +129,73 @@ func getTiDBInstances() []instance {
 	return []instance{}
 }
 
-// TODO: get total CPU use time through Prometheus.
-func getTotalCPUUseTime(component ComponentType, instances []instance, duration time.Duration) float64 {
-	return 1.0
+// this function assumes that addr is already a valid resolvable address
+func getInstanceNameFromAddress(addr string) string {
+	// In K8s, a StatefulSet pod address is composed of pod-name.peer-svc.namespace.svc:port
+
+	// Extract the hostname part without port
+	hostname := addr
+	portColonIdx := strings.LastIndex(addr, ":")
+	if portColonIdx >= 0 {
+		hostname = addr[:portColonIdx]
+	}
+
+	// Just to make sure it is not an IP address
+	ip := net.ParseIP(hostname)
+	if ip != nil {
+		// Hostname is an IP address, return the whole address
+		return addr
+	}
+
+	firstDotIdx := strings.Index(hostname, ".")
+	if firstDotIdx < 0 {
+		// Cannot infer pod name, return the whole address
+		return addr
+	}
+
+	return addr[:firstDotIdx]
+}
+
+func getInstanceNames(instances []instance) []string {
+	names := make([]string, 0, len(instances))
+	for _, inst := range instances {
+		names = append(names, getInstanceNameFromAddress(inst.address))
+	}
+	return names
 }
 
-// TODO: get total CPU quota through Prometheus.
-func getTotalCPUQuota(component ComponentType, instances []instance) uint64 {
-	return 1
+// TODO: suppport other metrics storage
+// get total CPU use time (in seconds) through Prometheus.
+func getTotalCPUUseTime(querier Querier, component ComponentType, instances []instance, timestamp time.Time, duration time.Duration) (float64, error) {
+	result, err := querier.Query(NewQueryOptions(component, CPUUsage, getInstanceNames(instances), timestamp, duration))
+	if err != nil {
+		return 0.0, err
+	}
```

Comment:
It seems we only extract the instance name and use it to link the store/tidb and prometheus targets. I think this would have some risk as we can deploy 2 tidbcluster with the same name in different namespace and make them as one united tidbcluster.

I think we also need to extract the namespace information.